text/3759-cargo-supported-targets.md

```diff
@@ -0,0 +1,494 @@
+even harder to represent, and things like `target_feature = "avx"` are basically not representable.
+Also, this is new syntax not currently used by cargo.
+
+### Allowing only target-triples
+
+This is an even stricter version of the above. Set relations between `supported-targets` lists are
+exact, and the resolver can determine if a platform-specific dependency can be pruned from the
+dependency tree more easily, hence why the original proposal chose this format. Being even simpler
+to implement, this alternative may not be expressive enough for the common use case. Packages rarely
+support specific target-triples, rather they support/require specific target attributes. What would
+likely happen is that packages would copy and paste the target-triple list matching their
+requirements from somewhere or someone else. Every time a new target with the same attribute is
+added, the whole ecosystem would have to be updated.
+
+# Prior art
+[prior-art]: #prior-art
+
+Users can already select which packages they want to select in a workspace with the flags
+`--package` and `--exclude`.
+
+The `per-package-target` nightly feature defines the `force-target` field, which is supposed to
+force the package to build for a specific target-triple. This does not interact well when used in
+dependencies, as one would expect a dependency to be built for the same target as the package.
+`supported-targets` supersedes `force-target` because instead of enforcing a single target, it
+enforces a set of targets.
+
+Published crates have mainly used their documentation to specify which targets they support, or they
+would leave it up to the user to infer it. Some crates also made use of compile time errors to
+ensure that `cfg` requirements are met, for example:
+```rust
+#[cfg(not(any(…)))]
+compile_error!("unsupported target cfg");
+```
+[`getrandom`](https://github.com/rust-random/getrandom/blob/9fb4a9a2481018e4ab58d597ecd167a609033149/src/backends.rs#L156-L160)
+is an example of a crate utilizing this method.
+
+In other system level languages, vendoring dependencies is a common practice, and the user would be
+responsible for ensuring that the dependencies are compatible with the target. For interpreted
+languages, this is a non-issue because any platform being able to run the interpreter can run the
+package.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Should we strip the `cfg` prefix from the field e.g., `supported-targets = 'target_os = "linux"'`?
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+## Ensuring proper use of dependencies
+
+By leveraging compilation errors, it would be possible to ensure that a package only uses
+dependencies that support the package's `supported-targets`. This would be done by comparing the
+`supported-targets` of the package with the `supported-targets` of the dependencies.
```

Comment:
imo, like the motivation section, this should focus on improving error messages and detection incompatible dependencies without exhaustively testing on every platform